tests/suite/graceful_recovery_test.go

```diff
@@ -98,8 +98,101 @@ var _ = Describe("Graceful Recovery test", Ordered, Label("functional", "gracefu
 	It("recovers when nginx container is restarted", func() {
 		runRecoveryTest(teaURL, coffeeURL, ngfPodName, nginxContainerName, files, &ns)
 	})
+
+	It("recovers when drained node is restarted", func() {
+		runRestartNodeWithDrainingTest(teaURL, coffeeURL, files, &ns)
+	})
+
+	It("recovers when node is restarted abruptly", func() {
+		runRestartNodeAbruptlyTest(teaURL, coffeeURL, files, &ns)
+	})
 })
 
+func runRestartNodeWithDrainingTest(teaURL, coffeeURL string, files []string, ns *core.Namespace) {
+	runRestartNodeTest(teaURL, coffeeURL, files, ns, true)
+}
+
+func runRestartNodeAbruptlyTest(teaURL, coffeeURL string, files []string, ns *core.Namespace) {
+	runRestartNodeTest(teaURL, coffeeURL, files, ns, false)
+}
+
+func runRestartNodeTest(teaURL, coffeeURL string, files []string, ns *core.Namespace, drain bool) {
+	nodeNames, err := getNodeNames()
+	Expect(err).ToNot(HaveOccurred())
+	Expect(nodeNames).To(HaveLen(1))
+
+	kindNodeName := nodeNames[0]
+
+	if portFwdPort != 0 {
+		close(portForwardStopCh)
+	}
+
+	if drain {
+		_, err := exec.Command(
+			"kubectl",
+			"drain",
+			kindNodeName,
+			"--ignore-daemonsets",
+			"--delete-local-data",
+		).CombinedOutput()
+		Expect(err).ToNot(HaveOccurred())
+
+		_, err = exec.Command("kubectl", "delete", "node", kindNodeName).CombinedOutput()
+		Expect(err).ToNot(HaveOccurred())
+	}
+
+	Expect(*clusterName).ToNot(BeEmpty())
+	containerName := *clusterName + "-control-plane"
+
+	_, err = exec.Command("docker", "restart", containerName).CombinedOutput()
+	Expect(err).ToNot(HaveOccurred())
+
+	// need to wait for docker container to restart and be running before polling for ready NGF Pods or else we will error
+	Eventually(
+		func() bool {
+			output, err := exec.Command(
+				"docker",
+				"inspect",
+				"-f",
+				"{{.State.Running}}",
+				containerName,
+			).CombinedOutput()
+			return strings.TrimSpace(string(output)) == "true" && err == nil
+		}).
+		WithTimeout(timeoutConfig.CreateTimeout).
+		WithPolling(500 * time.Millisecond).
+		Should(BeTrue())
+
+	// ngf can often oscillate between ready and error, so we wait for a stable readiness in ngf
+	var podNames []string
+	var count int
+	Eventually(
```

Comment:
[Consistently](https://pkg.go.dev/github.com/onsi/gomega#Consistently) would be a better fit than Eventually here.

Reply:
I was looking at the docs and saw:

> Consistently is useful in cases where you want to assert that something *does not happen* for a period of time.

But I also found the `MustPassRepeatedly` which does exactly I want. Do you think Consistently is still a better fit, given the case is that I want a ready NGF Pod name to be returned, and for it to pass repeatedly?

Reply:
I was unaware of `MustPassRepeatedly`. Looks like it's a better fit, nice find